scripts/linters/js_ts_linter.py

```diff
@@ -63,6 +63,31 @@
 ]
 
 
+def _parse_js_or_ts_file(filepath, file_content, **kwargs):
+    """Runs the correct function to parse the given file's source code.
+
+    With ES2015 and later, a JavaScript program can be either a script or a
+    module. It is a very important distinction, a parser such as Esprima needs
+    to know the type of the source to be able to analyze its syntax correctly.
+    This is achieved by choosing parseScript function to parse a script and
+    parseModule function to parse a module.
+
+    https://esprima.readthedocs.io/en/latest/syntactic-analysis.html#distinguishing-a-script-and-a-module
+
+    Args:
+        filepath: str. Path of the source file.
+        file_content: str. Code to compile.
+        **kwargs: dict(*: *). Passed along to esprima.
+
+    Returns:
+        dict. Parsed contents produced by esprima.
+    """
+    parse_function = (
+        esprima.parseScript if filepath.endswith('.js') else
+        esprima.parseModule)
```

Comment:
Should you do elif filepath endswith('.ts') else raise Exception, just as an additional safeguard?